metpy/calc/tools.py

```diff
@@ -1126,3 +1135,47 @@ def _process_deriv_args(f, kwargs):
         raise ValueError('Must specify either "x" or "delta" for value positions.')
 
     return n, axis, delta
+
+
+def parse_angle(input_dir_str, deg2rad=False):
+    """Calculate the meteorological angle from directional text
+
+    Works for abbrieviations or whole words (E -> 90 | South -> 180)
+    and also is able to parse 22.5 degreee angles such as ESE/East South East
+
+    Parameters
+    ----------
+    input_dir_str : string
+        Directional text such as west, south-west, ne, etc
+    deg2rad : boolean, optional
+        If true, returns a value in radian, else returns a value in degrees
+
+    Returns
+    -------
+    angle
+        The angle in degrees or radians
+    """
+    input_dir_str = _abbrieviate_direction(input_dir_str)
+    for i, dir_str in enumerate(DIR_STRS):
+        if input_dir_str.upper() == dir_str:
+            deg = i * 22.5
+            break
+
+    if deg2rad:
+        return np.deg2rad(deg)
+    else:
+        return deg
+
+
+def _abbrieviate_direction(ext_dir_str):
+    """Convert extended (non-abbrievated) directions to abbrieviation"""
+    return (ext_dir_str
```

Comment:
Would this be any easier to read:
```python
ext_dir_str = ext_dir_str.upper()
replacements = [('_', ''), ('-', ''), (' ', ''), ...]
for source, replace in replacements:
    ext_dir_str = ext_dir_str.replace(source, replace)
```
? My naming choices are likely terrible.

Reply:
I just tried this implementation, but I think I prefer the original method since the variable `replacements` becomes too long and there's no need for an extra for loop.

Reply:
Fair enough.